docs/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md

```diff
@@ -372,6 +373,30 @@ export default {
 }
 ```
 
+### Fields API
+
+The Fields API allows a plugin to replace a component within a field type by another one with `addFields()` function using the following parameters:
+
+| Parameter     | Type             | Description                                                                                                                                                                                                              |
+| ------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `type`          | String           | Type of field to customize |
+| `Component`        | React Component       | Component used for this type of field. <br /> Should be one of the content types available in the Content Type Builder:<ul><li>`string`</li><li>`text`</li><li>`richtext`</li><li>`wysiwyg`</li><li>`enumeration`</li><li>`email`</li><li>`password`</li><li>`uid`</li><li>`date`</li><li>`time`</li><li>`datetime`</li><li>`timestamp`</li><li>`integer`</li><li>`biginteger`</li><li>`float`</li><li>`decimal`</li><li>`boolean`</li><li>`array`</li><li>`json`</li><li>`media`</li><li>`relation`</li><li>`component`</li><li>`dynamiczone`</li></ul> |
```

Comment:
IMO the list of fields should be on the `type` section because the value of `type` must be one of the existing fields, WDYT?

Reply:
seems like a good idea indeed thank you 💡 
what do you think about it @pwizla?

Reply:
If this is what you mean 👇, sure! 
![Screenshot 2021-12-09 at 10 47 16](https://user-images.githubusercontent.com/4233866/145373075-2c652ffc-12cd-46e9-b844-b6f5dd902941.png)

Actually, judging from the example, `Component` should only be the name of the component to use, right? 🙂

Reply:
You can also add a link to the documentation for content-types (`/developer-docs/latest/development/backend-customization/models.html#model-attributes`) so people can get more information.

Reply:
Thanks @pwizla, that's exactly what I mean. We can even put this list in the type (replacing string) and not the description to be more precise in the typing. Indeed it is a string but in truth it is more specific than that. A string that is not in the list of fields will not be accepted. At least, with the current API.

Reply:
I am afraid that this will cause confusion for the developers the API is used more in order to add custom fields in the content manager and not customise an existing one even though in our guides we use it to customise the WYSIWYG. can you make an update to reflect this?

Reply:
I think I am missing some key elements to understand the API then, could we talk about it? 🙂

Reply:
Thanks @HichamELBSI 
I get what you mean. 
I think we should keep the "Type" column simple, so I would keep simply 'String' there, but we can be even more explicit in the description, then  :-)

We could rephrase the description like this:

"Type of field to customize.
  
  Should be one of the content-types available [link to doc] in the Content-Type Builder.
  Only accepts the following values:
  [insert list here]
  "
  
 So this would look a bit like what we did for other use cases, for example with the [hook event object](https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#hook-event-object) or [model settings](https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#model-settings) :-)